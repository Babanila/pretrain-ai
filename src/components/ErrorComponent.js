import React from "react";
import { cx, css } from "emotion";

function ErrorComponent({ message }) {
  const errorStyle = css`
    width: 100%;
    height: 100vh;
    overflow: hidden;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #f54b3a;
  `;

  return (
    <div className={cx(errorStyle)}>
      <h3>{!message ? "Page not found!!!" : message}</h3>
    </div>
  );
}

export default ErrorComponent;
