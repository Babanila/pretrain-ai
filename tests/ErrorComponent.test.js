import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ErrorComponent from '../src/components/ErrorComponent'

describe('Check ErrorComponent Component', () => {
  it('should renders without crashing', () => {
    const tree = renderer.create(<ErrorComponent />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct number of element in ErrorComponent component', () => {
    const ErrorComponentWrapper = shallow(<ErrorComponent />)
    const actual = ErrorComponentWrapper.find('div')

    expect(ErrorComponent).toBeDefined()
    expect(actual.exists).toBeTruthy()
    expect(actual.length).toEqual(1)
    expect(actual.text()).toEqual('Page not found!!!')
  })

  it('should render the message passed', () => {
    const ErrorComponentWrapper = shallow(<ErrorComponent message="Error occur" />)
    const actual = ErrorComponentWrapper.find('div')

    expect(ErrorComponent).toBeDefined()
    expect(actual.exists).toBeTruthy()
    expect(actual.length).toEqual(1)
    expect(actual.text()).toEqual('Error occur')
  })
})
