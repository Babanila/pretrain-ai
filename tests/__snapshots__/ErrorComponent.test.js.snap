// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check ErrorComponent Component should renders without crashing 1`] = `
<div
  className="css-bh2ae5"
>
  <h3>
    Page not found!!!
  </h3>
</div>
`;
